contributed_definitions/nyaml/NXopt.yaml

```diff
@@ -56,348 +43,675 @@ NXopt(NXobject):
+        - Characteristics of the medium surrounding the sample
+        - "Beam devices" for a beam path description
+        - Stages(NXmanipulator)
+        - Sensors and actuators to control or measure sample or beam properties
+      device_information(NXfabrication):
         doc: |
-          The name of the instrument.
-        \@version:
+          General device information of the optical spectroscopy setup, if
+          suitable (e.g. for a tabletop spectrometer or other non-custom build setups).
+          For custom build setups, this may be limited to the construction year.
+        exists: recommended
+        vendor:
+          exists: recommended
+        model:
+          exists: recommended
+        identifier:
+          exists: recommended
+        construction_year(NX_DATE_TIME):
+          exists: recommended
           doc: |
-            The used version of the hardware if available. If not a commercial
-            instrument use date of completion of the hardware.
-      company:
+            Datetime of setup's initial construction. This refers to the date of
+            first measurement after new construction  or relocation of the setup.
+            Should be a ISO8601 date/time stamp. It is recommended to add an explicit time zone,
+            otherwise the local time zone is assumed per ISO8601.
+      software_TYPE(NXprogram):
+        exists: recommended
+        program:
+          doc: |
+            Commercial or otherwise defined given name of the program that was
+            used to control any parts of the optical spectroscopy setup.
+            The uppercase TYPE should be replaced by a specification name, i.e.
+            "software_detector" or "software_stage" to specify the respective
+            program or software components.
+          \@version:
+            doc: |
+              Either version with build number, commit hash, or description of a
+              (online) repository where the source code of the program and build
+              instructions can be found so that the program can be configured in
+              such a way that result files can be created ideally in a
+              deterministic manner.
+          \@url:
+            exists: optional
+            doc: |
+              Description of the software by persistent resource, where the program,
+              code, script etc. can be found.
+      # general properties and reference frames
+      reference_frames(NXcoordinate_system_set):
+        # This should be extended later and hopefully replace angle_reference_frame
+        # and sample_normal_direction.
+        # Required reference frames:
+        # Sample normal, beam_z_axis, sample_stage
         exists: optional
         doc: |
-          Name of the company which build the instrument.
-      construction_year(NX_DATE_TIME):
+          Description of one or more coordinate systems that are specific to the
+          experiment.
+      angle_reference_frame:
```

Comment:
Why not use `NXcoordinate_system` or `NXtransformations`. If there is just an `NX_CHAR`, no tool can understand how the instrument is located in space. Maybe this is planned for later.

Reply:
Yea, this is actually a critical part and I have to discuss this in the teamforce meeting as I'm not aware how to solve this well in a long term run. This is for now only a short term solution.

Reply:
I think, the key issue is that we have no absolute coordinate system and have to start somewhere. If we use a text to define the convention, then NXtransformations can do the rest. Without them, we do not know what is the point of origin for the first NXtransformations.

Reply:
Yes, but it is exactly the purpose of NXcoordinate_system. @lukaspie suggests to use this rather than a generic NX_CHAR field. This we should take for further development after the OPT Workshop, but this PR can go without this

Reply:
We can do it later 👍